pkg/cortex/serve/init/bootloader.sh

```diff
@@ -52,44 +52,68 @@ if [ "$CORTEX_PROVIDER" != "local" ]; then
-    bash -e /mnt/project/dependencies.sh
-fi
 
-# install from conda-packages.txt
-if [ -f "/mnt/project/conda-packages.txt" ]; then
-    py_version_cmd='echo $(python -c "import sys; v=sys.version_info[:2]; print(\"{}.{}\".format(*v));")'
-    old_py_version=$(eval $py_version_cmd)
+    # execute script if present in project's directory
+    if [ -f "/mnt/project/dependencies.sh" ]; then
+        bash -e /mnt/project/dependencies.sh
+    fi
 
-    # look for packages in defaults and then conda-forge to improve chances of finding the package (specifically for python reinstalls)
-    conda config --append channels conda-forge
+    # install from conda-packages.txt
+    if [ -f "/mnt/project/conda-packages.txt" ]; then
+        py_version_cmd='echo $(python -c "import sys; v=sys.version_info[:2]; print(\"{}.{}\".format(*v));")'
+        old_py_version=$(eval $py_version_cmd)
 
-    conda install -y --file /mnt/project/conda-packages.txt
+        # look for packages in defaults and then conda-forge to improve chances of finding the package (specifically for python reinstalls)
+        conda config --append channels conda-forge
 
-    new_py_version=$(eval $py_version_cmd)
+        conda install -y --file /mnt/project/conda-packages.txt
 
-    # reinstall core packages if Python version has changed
-    if [ $old_py_version != $new_py_version ]; then
-        echo "warning: you have changed the Python version from $old_py_version to $new_py_version; this may break Cortex's web server"
-        echo "reinstalling core packages ..."
+        new_py_version=$(eval $py_version_cmd)
 
-        pip --no-cache-dir install \
-            -r /src/cortex/serve/serve.requirements.txt \
-            /src/cortex/serve/
-        if [ -f "/src/cortex/serve/image.requirements.txt" ]; then
-            pip --no-cache-dir install -r /src/cortex/serve/image.requirements.txt
+        # reinstall core packages if Python version has changed
+        if [ $old_py_version != $new_py_version ]; then
+            echo "warning: you have changed the Python version from $old_py_version to $new_py_version; this may break Cortex's web server"
+            echo "reinstalling core packages ..."
+
+            pip --no-cache-dir install \
+                -r /src/cortex/serve/serve.requirements.txt \
+                /src/cortex/serve/
+            if [ -f "/src/cortex/serve/image.requirements.txt" ]; then
+                pip --no-cache-dir install -r /src/cortex/serve/image.requirements.txt
+            fi
         fi
+    fi
 
-        rm -rf $CONDA_PREFIX/lib/python${old_py_version}  # previous python is no longer needed
+    # install pip packages
+    if [ -f "/mnt/project/requirements.txt" ]; then
+        pip --no-cache-dir install -r /mnt/project/requirements.txt
     fi
-fi
 
-# install pip packages
-if [ -f "/mnt/project/requirements.txt" ]; then
-    pip --no-cache-dir install -r /mnt/project/requirements.txt
+}
+
+# install user dependencies
+tempf=$(mktemp)
```

Comment:
As discussed on call, reorder this to stream output to stdout if cortex log level is debug or info and only buffer output if its warning or higher.